### PR TITLE
Test newly uploaded Flan-T5 weights

### PR DESCRIPTION
**Pull request by joecummings**
This PR adds tests for the Flan-T5 weights and confirms that `build_hf_checkpoint_from_path` works w/ Flan.

### CODE REVIEW

**Comment by joecummings** [comment]
This should be cherry-picked into the release as it's just more test coverage.

File: README.rst [hunk 1]
```diff
@@ -122,7 +122,7 @@ The library currently consist of following pre-trained models:
 * `DistilRoBERTa <https://github.com/huggingface/transformers/blob/main/examples/research_projects/distillation/README.md>`_
 * XLM-RoBERTa: `Base and Large Architure <https://github.com/pytorch/fairseq/tree/main/examples/xlmr#pre-trained-models>`_
 * T5: `Small, Base, Large, 3B, and 11B Architecture <https://github.com/google-research/text-to-text-transfer-transformer>`_
-* Flan-T5: `Small, Base, Large, XL, and XXL Architecture <https://github.com/google-research/t5x>`_
```
Comment (joecummings) [inline README.rst:125]: Don't actually support Flan-T5 small due to non divisible # of attention heads.

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -55,7 +55,6 @@ jobs:
         python3 -m pip --quiet install sentencepiece
         python3 -m pip --quiet install tqdm
         python3 -m pip --quiet install expecttest
-        python3 -m pip --quiet install transformers
```
Comment (joecummings) [inline .github/workflows/integration-test.yml:58]: No longer need transformers install for integration tests.

File: test/integration_tests/test_t5_models.py [hunk 3]
```diff
@@ -74,126 +80,72 @@ def _t5_model(self, is_jit, t5_model, expected_asset_name, test_text):
...
-            our_t5 = T5Bundle.build_model_from_huggingface_ckpt(model_path)
-
-            hf_output = t5_small_gen(
-                input_ids=self.encoder_input_ids,
-                decoder_input_ids=self.decoder_input_ids,
-                attention_mask=self.encoder_padding_mask,
-                decoder_attention_mask=self.decoder_padding_mask,
-                output_hidden_states=True,
-                output_attentions=True,
+        model = T5Bundle.build_model_from_huggingface_ckpt(model_path, encoder_only=is_encoder_only)
```
Comment (joecummings) [inline test/integration_tests/test_t5_models.py:142]: The above already tests the outputs of HF models, so we just need to confirm that we can load the weights from a HF file and that the model can run.

File: torchtext/models/t5/bundler.py [hunk 4]
```diff
@@ -238,12 +239,12 @@ def build_model_from_huggingface_ckpt(
 
             for i in range(config.num_decoder_layers):
                 if config.is_gated_act:
-                    t5_model_state_dict[f"encoder.layers.{i}.linear1_0.weight"] = hf_weights[
```
Comment (joecummings) [inline torchtext/models/t5/bundler.py:241]: This was a bug.

**Comment by Nayef211** [comment]
> This should be cherry-picked into the release as it's just more test coverage.

Could you create a release tracker issue similar to https://github.com/pytorch/text/issues/1766?

File: test/integration_tests/test_t5_models.py [hunk 5]
```diff
@@ -74,126 +80,72 @@ def _t5_model(self, is_jit, t5_model, expected_asset_name, test_text):
...
-                    # cross-attention scores
-                    assert torch.equal(
-                        our_output["decoder_ca_scores"][i], hf_output.cross_attentions[i]
-                    ), f"Mismatched cross-attention scores for decoder layer {i}"
-                # decoder hidden states
-                assert torch.equal(
-                    our_output["decoder_hidden_states"][i], hf_output.decoder_hidden_states[i]
-                ), f"Mismatched hidden states for decoder layer {i}"
-
-    def test_t5_bundler_load_hf_ckpt_pretrained_encoder_only(self) -> None:
```
Comment (Nayef211) [inline test/integration_tests/test_t5_models.py:133]: Why do we get rid of all of these tests?

**Review by Nayef211 — APPROVED** [review]
Overall LGTM

File: test/integration_tests/test_t5_models.py (hunk 5 above)
Comment (joecummings) [reply test/integration_tests/test_t5_models.py:133]: We do the actual testing of the weights in the above tests. Here we just want to make sure the code can be loaded and ran.

**Pull request merged**